### PR TITLE
Fix #1210: Coverity - Unlocked write at AuthenticationManagementService

### LINKED ISSUE #1210: Coverity - Unlocked write at AuthenticationManagementService
Reported at https://scan4.scan.coverity.com/reports.htm#v57086/p16071/fileInstanceId=126006193&defectInstanceId=14538870&mergedDefectId=382288
Spring documentation https://docs.spring.io/spring-security/site/docs/5.5.8/reference/html5/#servlet-authentication-securitycontextholder

### PR DESCRIPTION
**Pull request by banterCZ**

### CODE REVIEW

File: powerauth-webflow-authentication/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/service/AuthenticationManagementService.java [hunk 1]
```diff
@@ -181,8 +182,10 @@ public void authenticateCurrentSession() {
     public void pendingAuthenticationToAuthentication() {
         UserOperationAuthentication auth = getPendingUserAuthentication();
         if (auth.isAuthenticated()) {
-            logger.info("Security context was set to authenticated");
-            SecurityContextHolder.getContext().setAuthentication(auth);
+            final SecurityContext securityContext = SecurityContextHolder.createEmptyContext();
```
Comment (banterCZ) [inline powerauth-webflow-authentication/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/service/AuthenticationManagementService.java:185]: See Spring Security documentation https://docs.spring.io/spring-security/site/docs/5.5.8/reference/html5/#servlet-authentication-securitycontextholder

File: powerauth-webflow-authentication/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/service/AuthenticationManagementService.java [hunk 2]
```diff
@@ -181,8 +182,10 @@ public void authenticateCurrentSession() {
...
+            securityContext.setAuthentication(auth);
+            SecurityContextHolder.setContext(securityContext);
+            logger.info("UserOperationAuthentication(userId={}) set to to security context.", auth.getUserId());
```
Comment (banterCZ) [inline powerauth-webflow-authentication/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/service/AuthenticationManagementService.java:188]: May we log `userId`?

File: powerauth-webflow-authentication/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/service/AuthenticationManagementService.java [hunk 3]
```diff
@@ -181,8 +182,10 @@ public void authenticateCurrentSession() {
     public void pendingAuthenticationToAuthentication() {
         UserOperationAuthentication auth = getPendingUserAuthentication();
         if (auth.isAuthenticated()) {
-            logger.info("Security context was set to authenticated");
-            SecurityContextHolder.getContext().setAuthentication(auth);
+            final SecurityContext securityContext = SecurityContextHolder.createEmptyContext();
+            securityContext.setAuthentication(auth);
+            SecurityContextHolder.setContext(securityContext);
+            logger.info("UserOperationAuthentication(userId={}) set to the security context.", auth.getUserId());
```
Comment (banterCZ) [inline powerauth-webflow-authentication/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/service/AuthenticationManagementService.java:188]: May we log `userId`?

Comment (romanstrobl) [reply powerauth-webflow-authentication/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/service/AuthenticationManagementService.java:188]: Yes.

File: powerauth-webflow-authentication/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/service/AuthenticationManagementService.java (hunk 1 above)
Comment (romanstrobl) [reply powerauth-webflow-authentication/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/service/AuthenticationManagementService.java:185]: Ok.

**Review by romanstrobl — APPROVED** [review]
Nice improvements.

**Pull request merged**